synapse/push/bulk_push_rule_evaluator.py

```diff
@@ -113,15 +114,17 @@ def __init__(self, hs):
             resizable=False,
         )
 
-    async def _get_rules_for_event(self, event, context):
+    async def _get_rules_for_event(
+        self, event: EventBase, context: EventContext
+    ) -> Dict[str, List[Dict[str, Any]]]:
         """This gets the rules for all users in the room at the time of the event,
         as well as the push rules for the invitee if the event is an invite.
 
         Returns:
             dict of user_id -> push_rules
         """
         room_id = event.room_id
-        rules_for_room = self._get_rules_for_room(room_id)
+        rules_for_room = self._get_rules_for_room(room_id)  # type: ignore
```

Comment:
Why is this ignore necessary?

Reply:
Without it you get:

> synapse/push/bulk_push_rule_evaluator.py:127: error: Too few arguments for "_get_rules_for_room" of "BulkPushRuleEvaluator"  [call-arg]

I think the `lru_cache` decorator isn't being understood correctly.

Reply:
Ah, looks like we haven't added `_LruCachedFunction` to the mypy plugin to make `__call__` be "bound". Should be easy enough to add?

https://github.com/matrix-org/synapse/blob/cbc82aa09faa59acc20865e8b5c36561acb9a570/synapse/util/caches/descriptors.py#L119-L122

https://github.com/matrix-org/synapse/blob/cbc82aa09faa59acc20865e8b5c36561acb9a570/scripts-dev/mypy_synapse_plugin.py#L28-L36